src/Poker/Holdem/Table/DP.hs

```diff
@@ -0,0 +1,94 @@
+module Poker.Holdem.Table.DP
+  (
+    dpLookup
+  ) where
+
+
+import qualified Data.Array.Unboxed as Array
+import Data.Array.Base (unsafeAt)
+import GHC.Arr (unsafeIndex)
+
+
+dpLookup :: Array.UArray Int Int -> Int -> Int -> Int
+dpLookup q i k = dp2 `unsafeAt` idx
+  where idx = unsafeIndex ((0,0,0), (4,13,9)) (q `unsafeAt` i, 13 - i - 1, k)
+
+dp2 :: Array.UArray (Int, Int, Int) Int
+dp2 = Array.listArray ((0,0,0), (4,13,9)) [
```

Comment:
Not sure whether it matters, but this will all happen at runtime I'm pretty sure. If you want to precompile, you might be able to use TemplateHaskell to get even better runtime.